drone/init.lua

```diff
@@ -0,0 +1,73 @@
+local event = require("event")
+local component = require("component")
+local serialization = require("serialization")
+local port = require("configuration.constants").machineAddPort
+local IO = require("term")
+local modem = component.modem
+
+local droneData = {}
+
+local function save()
+    local file = io.open("/home/NIDAS/settings/droneData", "w")
+    file:write(serialization.serialize(droneData))
+    file:close()
+end
+
+local function queryLocation()
+    print("Drone location not set.")
+    IO.write("Insert X coordinate: ")
+    local x = tonumber(IO.read(_, true))
+    IO.write("Insert Y coordinate: ")
+    local y = tonumber(IO.read(_, true))
+    IO.write("Insert Z coordinate: ")
+    local z = tonumber(IO.read(_, true))
+    droneData.coordinates =  {x = x, y = y, z = z}
+    save()
+end
+
+local function load()
+    local file = io.open("/home/NIDAS/settings/droneData", "r")
+    if file ~= nil then
+        droneData = serialization.unserialize(file:read("*a"))
+        if droneData ~= nil then
+            if droneData.droneLocation ~= nil then
+                local x, y, z = component.navigation.getPosition()
+                if x == nil then error("The drone is out of range of the map.") end
+                local position = {x, y, z}
+                if droneData.droneLocation[1] ~= x or droneData.droneLocation[2] ~= y or droneData.droneLocation[3] ~= z then
```

Comment:
If the `location` in the file differs from the `droneLocation`, then the user has to guess the `droneLocation` again?